Mathlib/Analysis/Calculus/FDeriv/Mul.lean

```diff
@@ -570,7 +570,7 @@ theorem hasStrictFDerivAt_list_prod_finRange' {n : ℕ} {x : Fin n → 𝔸} :
 theorem hasStrictFDerivAt_list_prod_attach' [DecidableEq ι] {l : List ι} {x : {i // i ∈ l} → 𝔸} :
     HasStrictFDerivAt (𝕜 := 𝕜) (fun x ↦ (l.attach.map x).prod)
       (∑ i : Fin l.length, ((l.attach.take i).map x).prod •
-        smulRight (proj (l.attach.get (i.cast l.length_attach.symm)))
+        smulRight (proj (l.attach[i]'(by rw [l.length_attach]; exact i.2)))
```

Comment:
I guess the same question arises here; should `a[Fin.cast i h]` be preferred to `a[i]'(by something [h])`?

Reply:
I've encountered some awkwardness with `a[i]'(h)` when `i` is `Fin _` and `h` is nontrivial. I expect this to be fixed sooner than later but, since we happen to be in the present, I think `Fin.cast` has better support.

Reply:
My main concern is that `Finc.cast` is not the simp normal form here.

Reply:
That is a very legitimate concern. I think the simp normal form of `a[i]'(h)` is still fluid in some cases so it's a gamble either way.